golem-worker-executor-base/src/grpc.rs

```diff
@@ -668,6 +674,62 @@ impl<Ctx: WorkerCtx, Svcs: HasAll<Ctx> + UsesAllDeps<Ctx = Ctx> + Send + Sync +
             // TODO: add error details
         })
     }
+
+    async fn get_running_worker_metadatas_internal(
+        &self,
+        request: golem::workerexecutor::GetRunningWorkerMetadatasRequest,
+    ) -> Result<Vec<golem::worker::WorkerMetadata>, GolemError> {
+        let template_id: common_model::TemplateId = request
+            .template_id
+            .and_then(|t| t.try_into().ok())
+            .ok_or(GolemError::invalid_request("Invalid template id"))?;
+
+        let filter: Option<WorkerFilter> = match request.filter {
+            Some(f) => Some(f.try_into().map_err(GolemError::invalid_request)?),
+            _ => None,
+        };
+
+        let workers = self
+            .running_worker_enumeration_service()
+            .get(&template_id, filter)
+            .await?;
+
+        let result: Vec<golem::worker::WorkerMetadata> =
+            workers.iter().map(|worker| worker.clone().into()).collect();
+
+        Ok(result)
+    }
+
+    async fn get_worker_metadatas_internal(
+        &self,
+        request: golem::workerexecutor::GetWorkerMetadatasRequest,
+    ) -> Result<(Option<usize>, Vec<golem::worker::WorkerMetadata>), GolemError> {
+        let template_id: common_model::TemplateId = request
+            .template_id
+            .and_then(|t| t.try_into().ok())
+            .ok_or(GolemError::invalid_request("Invalid template id"))?;
+
+        let filter: Option<WorkerFilter> = match request.filter {
+            Some(f) => Some(f.try_into().map_err(GolemError::invalid_request)?),
+            _ => None,
+        };
+
+        let (new_cursor, workers) = self
+            .worker_enumeration_service()
+            .get(
+                &template_id,
+                filter,
+                request.cursor as usize,
+                request.count as usize,
```

Comment:
should we use `usize` or `u64`?

Reply:
we cannot have `usize` on the gRPC api so it's probably simpler to just use `u64` if possible. If the redis client requires `usize` then we have to have a conversion somewhere. 

`usize` is `u64` on 64bit architectures so it does not really matter.